### ISSUE TITLE
Illegal reflective operation warning

### ISSUE THREAD

**Issue by exu** [gradle/gradle#7445]
I have  warnings about __illegall reflective operation__ when running any gradle tasks

Gradle is in the most recent version available (4.10.2 - 2018-10-18)
It looks that it doesn't affect the behavior of builds and any other tasks. 

example output below:  

```
./gradlew -v

Welcome to Gradle 4.10.2!

Here are the highlights of this release:
 - Incremental Java compilation by default
 - Periodic Gradle caches cleanup
 - Gradle Kotlin DSL 1.0-RC6
 - Nested included builds
 - SNAPSHOT plugin versions in the `plugins {}` block

For more details see https://docs.gradle.org/4.10.2/release-notes.html

WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.codehaus.groovy.reflection.CachedClass (file:/home/exu/.gradle/wrapper/dists/gradle-4.10.2-all/9fahxiiecdb76a5g3aw9oi8rv/gradle-4.10.2/lib/groovy-all-2.4.15.jar) to method java.lang.Object.finalize()
WARNING: Please consider reporting this to the maintainers of org.codehaus.groovy.reflection.CachedClass
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release

------------------------------------------------------------
Gradle 4.10.2
------------------------------------------------------------

Build time:   2018-09-19 18:10:15 UTC
Revision:     b4d8d5d170bb4ba516e88d7fe5647e2323d791dd

Kotlin DSL:   1.0-rc-6
Kotlin:       1.2.61
Groovy:       2.4.15
Ant:          Apache Ant(TM) version 1.9.11 compiled on March 23 2018
JVM:          10.0.2 (Oracle Corporation 10.0.2+13)
OS:           Linux 4.18.14-arch1-1-ARCH amd64

                         
```

**Comment by oehme** [comment]
Duplicate of #6725

**Comment by mcandre** [comment]
Which Gradle release resolves this warning? I am using Gradle 5.1.1 with OpenJDK 11, Groovy 2.4.12, and still getting this behavior.
